sklearn/model_selection/tests/test_search.py

```diff
@@ -1846,3 +1839,68 @@ def test_search_cv__pairwise_property_equivalence_of_precomputed():
 
     attr_message = "GridSearchCV not identical with precomputed metric"
     assert (preds_original == preds_precomputed).all(), attr_message
+
+
+@pytest.mark.parametrize(
+    "SearchCV, param_search",
+    [(GridSearchCV, {'a': [0.1, 0.01]}),
+     (RandomizedSearchCV, {'a': uniform(1, 3)})]
+)
+def test_scalar_fit_param(SearchCV, param_search):
+    # unofficially sanctioned tolerance for scalar values in fit_params
+    # non-regression test for:
+    # https://github.com/scikit-learn/scikit-learn/issues/15805
+    class TestEstimator(BaseEstimator, ClassifierMixin):
+        def __init__(self, a=None):
+            self.a = a
+
+        def fit(self, X, y, r=None):
+            self.r_ = r
+
+        def predict(self, X):
+            return np.zeros(shape=(len(X)))
+
+    model = SearchCV(TestEstimator(), param_search)
+    X, y = make_classification(random_state=42)
+    model.fit(X, y, r=42)
+    assert model.best_estimator_.r_ == 42
+
+
+def _custom_lgbm_metric(y_test, y_pred):
+    # y_pred are probablities which need to be thresholded
+    y_pred = (y_pred > 0.5).astype(int)
+    acc = accuracy_score(y_test, y_pred)
+    # required output of format: (eval_name, eval_result, is_higher_better)
+    return ('accuracy', acc, True)
+
+
+@pytest.mark.parametrize("metric", ['auc', _custom_lgbm_metric])
+@pytest.mark.parametrize(
+    "SearchCV, param_search",
+    [(GridSearchCV, {'learning_rate': [0.1, 0.01]}),
+     (RandomizedSearchCV, {'learning_rate': uniform(0.01, 0.1)})]
+)
+def test_scalar_fit_param_lightgbm(metric, SearchCV, param_search):
```

Comment:
I guess we can simply remove this test (and lightgbm dependency)

Reply:
No because we already use `lightgbm` for some tests in `HistGradientBoosting*`and this could be handy to detect any breaking change.

Reply:
I don't agree. I don't think lightgbm happening to be on the receiving end of these fit_params tells us anything more. If we want to test for breaking changes with external software, we or they need to develop extensive integration test suites. Here we are trying to test that a specific feature of scikit-learn works as advertised, towards which the specific behaviours of lightgbm are irrelevant.
The use of lightgbm in HistGradientBoosting* estimators is specifically necessitated and does not set precedent for its inclusion here, IMO.